crates/bifrost/src/providers/replicated_loglet/network.rs

```diff
@@ -161,8 +163,41 @@ impl RequestPump {
         provider: &ReplicatedLogletProvider<T>,
         incoming: Incoming<Append>,
     ) {
+        let current_logs_version = provider.networking().metadata().logs_version();
+
+        let logs_version = incoming
+            .metadata_version()
+            .logs
+            .unwrap_or(current_logs_version);
+
         let (reciprocal, append) = incoming.split();
 
+        if logs_version > current_logs_version {
+            // if logs version is still in the future
+            // sync to this version (or newer)
+            if let Err(err) = provider
+                .networking()
+                .metadata()
+                .sync(MetadataKind::Logs, TargetVersion::Version(logs_version))
+                .await
+            {
+                match err {
+                    SyncError::Shutdown(_) => {
+                        return_error_status!(reciprocal, SequencerStatus::Shutdown);
+                    }
+                    SyncError::MetadataStore(err) => {
+                        return_error_status!(
+                            reciprocal,
+                            SequencerStatus::Error {
+                                retryable: true,
+                                message: err.to_string()
+                            }
+                        );
+                    }
+                }
+            }
+        }
```

Comment:
I suggest we only do this if we couldn't find the loglet/segment locally to avoid unnecessary syncs triggered by appends of unrelated loglets.  Also, let's keep retrying on metadata store errors (with delay).